internal/config/config.go

```diff
@@ -88,25 +91,53 @@ func SetNucleusConfig(config *NucleusConfig) error {
 	return nil
 }
 
-// Get or Creates the Nucleus folder that lives in the homedir and stores persisted settings
-func GetOrCreateNucleusFolder() (string, error) {
-	dirname, err := os.UserHomeDir()
-	if err != nil {
-		return "", err
-	}
-
-	fullName := dirname + "/" + nucleusFolderName
-
-	_, err = os.Stat(fullName)
+// ensureDirectoryExists checks for directory existence and tries to create it if it does not exist.
+func ensureDirectoryExists(dirName string) error {
+	_, err := os.Stat(dirName)
 	if os.IsNotExist(err) {
-		err = os.Mkdir(fullName, 0755)
+		err = os.Mkdir(dirName, 0755)
 		if err != nil {
 			if os.IsExist(err) {
-				return fullName, nil
+				return nil
 			}
-			return "", err
+			return err
 		}
 	} else if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Get or Creates the Nucleus folder that lives and stores persisted settings.
+//
+// 1. Checks for directory specified by env var NUCLEUS_CONFIG_DIR
+// 2. Checks for existence of XDG_CONFIG_HOME and append "nucleus" to it, if exists
+// 3. Use ~/.nucleus
+func GetOrCreateNucleusFolder() (string, error) {
+	configDir := os.Getenv("NUCLEUS_CONFIG_DIR")  // helpful for tools such as direnv and people who want it somewhere interesting
```

Comment:
if `viper.GetString` is used here, that will tap in to the env management system we have setup with the CLI.
that will for example let someone set this in the settings config and have it persist.

Reply:
., if we look up the config file based in root based on NUCLEUS_CONFIG_DIR and allow overriding it there.. not sure what the point of all of that would be.

Reply:
Hm, maybe I'm missing something.
Isn't the addition of `NUCLEUS_CONFIG_DIR` here giving the user the option to override the default location of where the nucleus folder is?

Reply:
Yeah, that's what I'm trying to do. I don't think it should be a setting in a configuration file as well, just b/c we're using it and then overriding it, which seems surprising to me. Also unsure if that would add value.. but maybe i'm not understanding something?

Reply:
oh, well I was thinking that would be a value add.

Today we default the nucleus folder to just be in $HOME as it's hardcoded here.

However, you're adding in the ability to allow someone to specify a `NUCLEUS_CONFIG_DIR` as well as honor `XDG_CONFIG_HOME` environment variable to override where the default place is for this nucleus folder.

The CLI uses viper as it's config + environment variable management system.
So I'm just wondering why not let viper know about these two new env vars, so that it can tap in to this.

We let people provide a `.nucleus-cli.yaml` file today, but viper is only smart enough to look in ~/.nucleus, `~/` and `.`.
So if someone provides this new `NUCLEUS_CONFIG_DIR` env var, it won't be smart enough to find the `.nucleus-cli.yaml` file there. same goes with `XDG_CONFIG_HOME`.

You making this change had me wondering:

We could add these two new env vars into the Viper system. 
Then, when it comes time to actually upsert the nucleus folder to create the `auth.yaml` file, we can just ask viper where the nucleus_config_dir is supposed to be. If nobody provides it, then we can just set the default value for `NUCLEUS_CONFIG_DIR` to `~/.nucleus` (or `$XDG_CONFIG_HOME/.nucleus`)

Reply:
I agree the paths should be added to viper startup.. i don't use .nucleus-cli.yaml (i use direnv instead of much the same purpose currently) so had forgotten about that.

I don't much see the point of the other changes and certainly don't want them to change after they're read from a config file.

Reply:
sorry yea not suggesting we allow someone to set the NUCLEUS_CONFIG_DIR or XDG_CONFIG_HOME from the config file.

Maybe adding them to the viper startup paths is sufficient. Then using os.Getenv here works as expected instead of using viper.Getstring, which might cause some weirdness if someone tries to override it in strange places.

Reply:
> sorry yea not suggesting we allow someone to set the NUCLEUS_CONFIG_DIR or XDG_CONFIG_HOME from the config file.

however I kind of was when asking if we could use viper.Getstring. But it all makes sense to me now. os.Getenv is going to sidestep all of that computation.